src/js/helpers/graph/declarationHelper.js

```diff
@@ -0,0 +1,150 @@
+/*
+ * Copyright (C) 2020 Knot.x Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import GraphLayers from './graphLayers';
+
+const getNodeGroup = (node) => {
+  if (node.type === 'virtual_start') {
+    return 'virtual';
+  }
+
+  if (node.type === 'virtual_end') {
+    return `virtual_${node.status}`;
+  }
+
+  return node.status;
+};
+
+const isComposite = (node) => node.type === 'composite';
+
+const hasTransitions = (node) => !!node.on;
+
+const createVisNode = (node) => ({
+  id: node.id,
+  label: node.label,
+  group: getNodeGroup(node),
+});
+
+const getLeaves = (root, depth = 0) => {
+  if (!root.on) {
+    return { node: root, depth };
+  }
+
+  return Object.entries(root.on)
+    .flatMap(([, child]) => getLeaves(child, depth + 1))
+    .sort((a, b) => b.depth - a.depth);
+};
+
+const createTransitionTo = (node, name) => ({ name, node, isReference: typeof node === 'string' });
+
+const getTransitions = (node) => {
+  if (hasTransitions(node)) {
+    return Object.entries(node.on)
+      .map(([transitionName, child]) => createTransitionTo(child, transitionName));
+  }
+
+  return [];
+};
+
+const createVirtualStartNode = (composite, transitions) => ({
+  ...composite,
+  label: '',
+  type: 'virtual_start',
+  id: `${composite.id}_virtual`,
+  on: transitions,
+});
+
+const createVirtualEndNode = (composite, transitions) => ({
+  ...composite,
+  label: '',
+  type: 'virtual_end',
+  id: `${composite.id}_virtual2`,
+  on: transitions,
+});
+
+const getTransitionsFromSubtasks = (subtasks) => subtasks
+  .reduce((total, current, index) => ({ ...total, [`_subtask_${index}`]: current }), {});
```

Comment:
Those transitions connect the virtual node with subtasks. So we create a structure like:
```
{
  _subtask_0 : subtaskNode,
  _subtask_1 : nextSubtaskNode,
  _subtask_2 : nextNextSubtaskNode
}
```
Those transitions are not "from", more "to". Maybe we should also rename this method to "create...".